### PR TITLE
Add git hook to ask about sensitive data before pushes.

### PR DESCRIPTION
**Pull request by surajreddy**
### What it does and why:
- Adds a git hook script that can be copied to prompt you to check for sensitive data before pushes.
- Run the script `git_hooks/create_hooks` to create a symlink for this hook in your `./.git/hooks/` folder.

### Checklist:

- [X] Testing & QA: Passes tests and linting.
- [X] :tophat:: I have manually tested these changes.

### CODE REVIEW

**Comment by JordanVerasamy** [comment]
👍  🎩

Should we add something to the README letting people know that if they clone this repo they should run `./git_hooks/create_hooks`?

**Comment by mimperatore** [comment]
Client-side hooks are still going to be prone to accidental leakage if someone forgets to set them up.

Have we considered a pre-receive server-side hook, coupled with a requirement to specify a given value to the `-o/--push-option` (e.g. `-o 'no sensitive data'`)?

File: git_hooks/create_hooks [hunk 1]
```diff
@@ -0,0 +1,3 @@
+#!/bin/bash
+HOOK_DIR=$(git rev-parse --show-toplevel)/.git/hooks
+ln -s -f ../../git_hooks/pre-push $HOOK_DIR/pre-push
```
Comment (ariens-shopify) [inline git_hooks/create_hooks]: Newline?

Comment (JordanVerasamy) [reply git_hooks/create_hooks]: dont really agree, how strongly do you feel about this

**Comment by mimperatore** [comment]
> Should we add something to the README letting people know that if they clone this repo they should run ./git_hooks/create_hooks?
@JordanVerasamy Just noticed this... yes, I think it's a good idea to do so.

**Pull request merged**